src/objectExplorer/objectExplorerService.ts

```diff
@@ -135,7 +135,8 @@ export class ObjectExplorerService {
                 const credentials = self._sessionIdToConnectionCredentialsMap.get(result.sessionId);
                 const children = result.nodes.map(node => TreeNodeInfo.fromNodeInfo(node, result.sessionId,
                     self._currentNode, credentials));
-                self._treeNodeToChildrenMap.set(self._currentNode, children);
+                let parentNode = self._rootTreeNodeArray.find(n => n.nodePath === result.nodePath);
```

Comment:
You're still setting the parent node of the children to self._currentNode which doesn't seem correct